Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md

```diff
@@ -294,3 +294,126 @@ angular.module("umbraco").controller("customBlockController", function ($scope,
+            GuidUdi contentUdi2 = new GuidUdi("element", System.Guid.NewGuid());  //Since these will be BLocklist items the Guids need to mention the "element" keyword
+            IContent request = contentService.Create("Home", -1, "home", -1);  //using Content Service, we create a new root node with the name Home and Document Type home
+            Person person1 = new Person("Janice", "Janice@janiceindustries.com"); 
+            Person person2 = new Person("John", "john@johnindustries.com");  //we create some dummy users
+            Blocklist blocklistNew = new Blocklist();  //initialize our new empty model to mimic proper JSON structure
+            var personList = new List<Dictionary<string, string>>();  //initialize empty person list where we will add our users
+            var contentTypes = contentTypeService.GetAll();  //we get the content types to later get the Person Document Type key from
+            var personType = contentTypes.Where(n => n.Alias == "person").FirstOrDefault();  //using the above types, we locate the one that corresponds to Person Document Type
+            var dictionaryUdi = new List<Dictionary<string, string>>();  //the dictionaryUdi list here will be passed in the first section of our final JSON object
+
+
+            personList.Add(new Dictionary<string, string> //add person1
+            {
+                {"contentTypeKey", personType.Key.ToString()},  //we need to pass the key of the Blocklist item type, we used ContentTypeService to obtain it.
+                {"udi", contentUdi1.ToString()},  //each item should also have a unique udi. We are passing the one we generated before
+                {"user_name", person1.user_name},  //document type custom property
+                {"user_email", person1.user_email}  //document type custom property
+            });
+            dictionaryUdi.Add(new Dictionary<string, string> { { "contentUdi", contentUdi1.ToString() } });  //with person 1 added to the contentData section of our JSON, we also add a reference to layout section
+            personList.Add(new Dictionary<string, string> //add person2
+            {
+                {"contentTypeKey", personType.Key.ToString()},
+                {"udi", contentUdi2.ToString()},
+                {"user_name", person2.user_name},
+                {"user_email", person2.user_email}
+            });
+            dictionaryUdi.Add(new Dictionary<string, string> { { "contentUdi", contentUdi2.ToString() } });
+
+
+            blocklistNew.layout = new BlockListUdi(dictionaryUdi);  //first section of JSON must contain udi references to whatever is in contentData
+            blocklistNew.contentData = personList;  //contentData is a list of our Person objects
+            blocklistNew.settingsData = new List<Dictionary<string, string>>();  //we initialize settingsData with a new empty List so it does not return null
```

Comment:
Here the comment doesn't explain what the settingsData is for, or why we create an empty list, eg to avoid it returning null yes, but if you are new to this then why?
```suggestion
            blocklistNew.settingsData = new List<Dictionary<string, string>>();  //If your Block List Item has settings, you would use the same technique as for contentData to set the setting values here, based on the underlying settings Document Type. If there are no settings for the item, we still need to set an empty list here to avoid a null error.
```